protocol/communities/manager.go

```diff
@@ -360,7 +374,80 @@ func (m *Manager) runENSVerificationLoop() {
 					return
 				}
 				m.logger.Info("received records", zap.Any("records", records))
+			}
+		}
+	}()
+}
+
+func (m *Manager) runOwnerVerificationLoop() {
+	m.logger.Info("starting owner verification loop")
+	go func() {
+		for {
+			select {
+			case <-m.quit:
+				m.logger.Debug("quitting owner verification loop")
+				return
+			case <-time.After(validateInterval):
+				// If ownerverifier is nil, we skip, this is useful for testing
+				if m.ownerVerifier == nil {
+					continue
+				}
+
+				communitiesToValidate, err := m.persistence.getCommunitiesToValidate()
+
+				if err != nil {
+					m.logger.Error("failed to fetch communities to validate", zap.Error(err))
+					continue
+				}
+				for id, communities := range communitiesToValidate {
+					m.logger.Info("validating communities", zap.String("id", id), zap.Int("count", len(communities)))
+
+					for _, communityToValidate := range communities {
+						signer, description, err := m.unwrapCommunityDescriptionMessage(communityToValidate.payload)
+						if err != nil {
+							m.logger.Error("failed to unwrap community", zap.Error(err))
+							continue
+						}
+
+						chainID := CommunityDescriptionTokenOwnerChainID(description)
+						if chainID == 0 {
+							// This should not happen
+							m.logger.Error("chain id is 0, ignoring")
+							continue
+						}
+
+						m.logger.Info("validating community", zap.String("id", types.EncodeHex(communityToValidate.id)), zap.String("signer", common.PubkeyToHex(signer)))
 
+						ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
```

Comment:
Is this a 3 seconds "sleep"? What is it for?

Reply:
It is a timeout for the contract call.